### PR TITLE
Guest page table

### PR DESCRIPTION
**Pull request by mingqiangchi**

### CODE REVIEW

File: arch/x86/guest/guest.c [hunk 1]
```diff
@@ -335,3 +335,103 @@ int prepare_vm0_memmap_and_e820(struct vm *vm)
...
+	 */
+	sub_table_addr = (uint64_t)_ld_guest_initial_page_table_start
+		+ PAGE_SIZE_4K;
+	entry = sub_table_addr | table_present;
+	MEM_WRITE64(pml4_addr, entry);
+
+	/* Write PDPTE, PDPT used 1 page
+	 * skip 2 pages to fetch PD
+	 */
+	sub_table_addr = (uint64_t)_ld_guest_initial_page_table_start
```
[the remark this answers is not in the document]
Comment (JasonChenCJ) [reply arch/x86/guest/guest.c]: maybe we can create multi variants of sub_table_addr, like l3_table_paddr_base, l2_table_paddr_base?

**Comment by dongyaozu** [comment]
+      uint64_t entry = 0;

+      uint64_t entry_num = 0;

+      uint64_t sub_table_addr = 0;

+      uint64_t table_present = 0;

+      uint64_t table_offset = 0;

+      uint64_t pml4_addr = (uint64_t)GPA2HVA(vm,

+                     (uint64_t)_ld_guest_initial_page_table_start);

+

+      /* Write PML4E */

+      table_present = (IA32E_COMM_P_BIT

+                     | IA32E_COMM_RW_BIT);

+      /* Using continuous memory for guest page tables,

+       * PML4 used 1 page, skip it to fetch PDPT

+       */

+      sub_table_addr = (uint64_t)_ld_guest_initial_page_table_start

@dongyaozu<https://github.com/dongyaozu> , here _ld_guest_initial_page_table_start we recognize it as gpa.





This is a symbol in LDS file, right? If this is the case, it is VA.

**Comment by JasonChenCJ** [comment]
@dongyaozu agree with you, we will try to figure out another way.

**Comment by mingqiangchi** [comment]
Jason submit another PR

**Pull request closed without merging**